### PR TITLE
Aik/reduce noise

### PR DESCRIPTION
**Pull request by Wappa**
Hello Bro I fix all my problem with grayscale I was completely die yesterday. Well you just have to import grayscale in the main take a image on internet and run. I create a key bind it load the bmp without grayscale. If you want want to grayscale the picture you need to press "g". I something go wrong you can comment this pull request
This function use a parameter char filepath were you put the path where is your image.

Please be nice accept my pull request

### CODE REVIEW

**Comment by yaneris** [comment]
So now you have just one function, like my load.c before ? So we have to call our future functions in your grayscale.c.

**Comment by yaneris** [comment]
PR accepted from me but needed @AurelReb 's approval.

File: src/preprocessing/grayscale.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+{
+    int quit = 0;
+    SDL_Event event;
+ 
+    SDL_Init(SDL_INIT_VIDEO);
+    IMG_Init(IMG_INIT_JPG);
+ 
+    SDL_Window * window = SDL_CreateWindow("SDL2 Grayscale",
+        SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 640, 480, 0);
+    SDL_Renderer * renderer = SDL_CreateRenderer(window, -1, 0);
```
Comment (AurelReb) [inline src/preprocessing/grayscale.c:15]: Can you write the `*` of the pointers right before the variable name ?: ` SDL_Renderer *renderer = ...`
Without this syntax, we could think it's the multiplier operator.

**Comment by AurelReb** [comment]
Please check the pointers' syntax

**Review by AurelReb — CHANGES_REQUESTED** [review]

File: src/preprocessing/grayscale.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    SDL_Surface *originalImage = image;
+    image = SDL_ConvertSurfaceFormat(image, SDL_PIXELFORMAT_ARGB8888, 0);
+    SDL_FreeSurface(originalImage);
+    Uint32 *pixels = (Uint32 *)image->pixels;
+ 
+  
+    while (!quit)
+    {
+        SDL_UpdateTexture(texture, NULL, image->pixels,
+    image->w *sizeof(Uint32));
```
Comment (AurelReb) [inline src/preprocessing/grayscale.c:31]: I think this is not a pointer ;) leave it as a multiplication

**Review by AurelReb — CHANGES_REQUESTED** [review]

**Review by AurelReb — APPROVED** [review]

**Pull request merged**